src/cmap/auth/mongodb_aws.ts

```diff
@@ -34,6 +57,7 @@ interface AWSSaslContinuePayload {
 }
 
 export class MongoDBAWS extends AuthProvider {
+  static credentialProvider: ReturnType<typeof getAwsCredentialProvider> | null = null;
```

Comment:
This serves two purposes, 1. we usually cache our lazy imports so we don't call require more than once, 2. I can now access the SDK this auth code uses for testing purposes

Packages (this includes the driver!) that compile with typescript have all the properties on their `module.exports` defined as non-configurable getters. While this emulates ES Module live binding and immutability, it makes sinon spying impossible.